lib/concurrent-ruby/concurrent/utility/processor_counter.rb

```diff
@@ -36,17 +36,22 @@ def compute_physical_processor_count
               when /darwin\d\d/
                 IO.popen("/usr/sbin/sysctl -n hw.physicalcpu", &:read).to_i
               when /linux/
-                cores = {} # unique physical ID / core ID combinations
-                phy   = 0
-                IO.read("/proc/cpuinfo").scan(/^physical id.*|^core id.*/) do |ln|
-                  if ln.start_with?("physical")
-                    phy = ln[/\d+/]
-                  elsif ln.start_with?("core")
-                    cid        = phy + ":" + ln[/\d+/]
-                    cores[cid] = true if not cores[cid]
+                # https://kernel.googlesource.com/pub/scm/linux/kernel/git/glommer/memcg/+/cpu_stat/Documentation/cgroups/cpu.txt
+                if Dir.exist?("/sys/fs/cgroup/cpu,cpuacct") && (cfs_quota_us = File.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us").to_i) > 0
```

Comment:
IMO we shouldn't change the behavior of `physical_processor_count`. We should instead have a `cpu_quota` method that returns a float, and then eventually a `usable_processor_count` which returns `cpu_quota * processor_count`.

This way it's up to the caller to decide if they want to `floor/round/ceil`

Reply:
Also I think what you implemented here is cgroups V1? Not sure how much it's used these days, in cgroups V2 the info is in `/sys/fs/cgroup/cpu.max`.